ckb-bin/src/subcommand/run.rs

```diff
@@ -192,33 +193,38 @@ fn verify_genesis(shared: &Shared) -> Result<(), ExitCode> {
 
 fn check_spec(shared: &Shared, args: &RunArgs) -> Result<(), ExitCode> {
     let store = shared.store();
-    if let Some(spec_hash) = store.get_chain_spec_hash() {
-        if args.chain_spec_hash != spec_hash && !args.skip_chain_spec_check {
+    if args.override_chain_spec {
+        write_chain_spec_hash(store, &args.chain_spec_hash)?;
+    } else if store.get_chain_spec_hash().is_none() {
+        write_chain_spec_hash(store, &args.chain_spec_hash)?;
+    } else if !args.skip_chain_spec_check {
+        let stored_spec_hash = store.get_chain_spec_hash().expect("checked above");
+        if args.chain_spec_hash != stored_spec_hash {
             eprintln!(
-                "chain_spec_hash mismatch Config({}) storage({}), pass command line argument --skip-spec-check if you are sure that the two different chains are compatible.",
-                args.chain_spec_hash, spec_hash
+                "chain_spec_hash mismatch Config({}) storage({}), pass command line argument \
+                --skip-spec-check if you are sure that the two different chains are compatible; \
+                or pass --override-spec to force overriding stored chain spec with configured chain spec",
+                args.chain_spec_hash, stored_spec_hash
             );
             return Err(ExitCode::Config);
         }
-    } else {
```

Comment:
Maybe u could just keep the old `if ... else ...` and change this `else` to `else if args.override_chain_spec`.

Reply:
When `args.chain_spec_hash != spec_hash`, it may enter the [`!args.skip_chain_spec_check branch`](https://github.com/nervosnetwork/ckb/pull/2505/files#diff-24827db4c0deb64085aac4223696a31814046c5cc185bc289124a77d8a9a01fbL196).

Simply changing `else` to `else if` is not enough.

I have updated the code, please help review again, thanks.